pkg/resource/deploy/source_query.go

```diff
@@ -99,10 +116,15 @@ func (src *querySource) Wait() result.Result {
 	}
 
 	select {
-	case src.res = <-src.finChan:
+	case src.res = <-src.langPluginFinChan:
 		// Language plugin has exited. No need to call `Close`.
 		src.done = true
 		return src.res
+	case src.res = <-src.providerRegErrChan:
+		// Provider registration has failed.
+		src.done = true
```

Comment:
Nit: this isn't necessary--the call to `src.Close()` will cover this.

Reply:
Let's remove it from the case below, too, then.